### DISCUSSION TITLE
Can emscripten compile itself (or gcc)? >:)

### DISCUSSION THREAD

**Discussion by NovaAndrom3da** [emscripten-core/emscripten#16354] [Q&A]
I want to be able to compile c/c++ from *within* the browser itself, and be able to successfully load it as wasm. Is it possible to do so by compiling itself or a c/c++ compiler? Thanks.

**Comment by NovaAndrom3da** [comment]
I just saw the second answer in https://github.com/emscripten-core/emscripten/discussions/16329 mentioning emception. Could this possibly be a way to do so?

**Reply by kripken** [reply]
Yes, emception does exactly that. See also [cib](https://tbfleming.github.io/cib/) (older, not sure if mtaintained).
